src/EditorFeatures/CSharpTest/Formatting/Indentation/CSharpFormatterTestsBase.cs

```diff
@@ -80,7 +81,7 @@ private static async Task TokenFormatWorkerAsync(TestWorkspace workspace, ITextB
 
             var formattingRuleProvider = workspace.Services.GetService<IHostDependentFormattingRuleFactoryService>();
 
-            var rules = formattingRuleProvider.CreateRule(document, position).Concat(Formatter.GetDefaultFormattingRules(document));
+            var rules = ImmutableArray.Create(formattingRuleProvider.CreateRule(document, position)).AddRange(Formatter.GetDefaultFormattingRules(document));
```

Comment:
could the `IHostDependentFormattingRuleFactoryService` return an immutable array?  or does it need to be ienumerable for some consumer

Reply:
teh host actually returns a single rule.  but then we need to wrap that in an immutable array to be able to addrange after that.  it's def not clean and i wish tehre was a better idiom for "i want to make an immutable array out of individual objects and streams of objects".